src/internal/connector/exchange/exchange_data_collection_test.go

```diff
@@ -104,3 +106,100 @@ func (suite *ExchangeDataCollectionSuite) TestExchangeCollection_AddJob() {
 
 	suite.Equal(len(shopping), len(eoc.jobs))
 }
+
+func (suite *ExchangeDataCollectionSuite) TestNewCollection_state() {
+	fooP, err := path.Builder{}.
+		Append("foo").
+		ToDataLayerExchangePathForCategory("t", "u", path.EmailCategory, false)
+	require.NoError(suite.T(), err)
+	barP, err := path.Builder{}.
+		Append("bar").
+		ToDataLayerExchangePathForCategory("t", "u", path.EmailCategory, false)
+	require.NoError(suite.T(), err)
+
+	table := []struct {
+		name   string
+		prev   path.Path
+		curr   path.Path
+		expect data.CollectionState
+	}{
+		{
+			name:   "new",
+			curr:   fooP,
+			expect: data.NewState,
+		},
+		{
+			name:   "not moved",
+			prev:   fooP,
+			curr:   fooP,
+			expect: data.NotMovedState,
+		},
+		{
+			name:   "moved",
+			prev:   fooP,
+			curr:   barP,
+			expect: data.MovedState,
+		},
+		{
+			name:   "deleted",
+			prev:   fooP,
+			expect: data.DeletedState,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			c := NewCollection(
+				"u",
+				test.curr, test.prev,
+				0, nil, nil, control.Options{})
+			assert.Equal(t, test.expect, c.State())
+		})
+	}
+}
+
+func (suite *ExchangeDataCollectionSuite) TestStateOf() {
```

Comment:
this test is not particularly useful given that `TestNewCollection_state` does the same thing, just less directly. I would be alright with removing it as it tests an unexported function that is tested by other stuff